SPECS/mariner-rpm-macros/macros

```diff
@@ -32,6 +32,7 @@
 # are consistently applied this override should be removed.
 %__spec_build_template	#!%{__spec_build_shell}\
 %{__spec_build_pre}\
+/usr/lib/rpm/mariner/gen-ld-script.sh\
```

Comment:
is `%{mariner_release_version}` populated here? Possible enhancement might be to plumb that into the script and dynamically populate.

I've got some code somewhere that does hex translation in bash, I'll dig it up.

Reply:
Found it, was almost perfectly portable. Modified it to fit this use quickly.

```
#!/bin/bash

version=$1
[[ $version =~ ^[0-9]\.[0-9]$ ]] || { echo "$version is not a supported version"; exit 1; }

version_major=${version%%.*}
version_minor=${version##*.}

#Output digit without newline, od without address info (-An), and of type single byte hex (-t x1)
hex_major="0x$(printf $version_major | od -An -t x1 | xargs)"
hex_minor="0x$(printf $version_minor | od -An -t x1 | xargs)"

echo "BYTE($hex_major) BYTE(0x2E) BYTE($hex_minor)"
```

```
./release_test.sh 1.0
BYTE(0x31) BYTE(0x2E) BYTE(0x30)

./release_test.sh 134.456
123.456 is not a supported version (exit 1)
```

Reply:
Thanks for sharing. Yes, ${mariner_release_version} is available here and I was able to pass it to the script. However, I plan to add these enhancements later on. Adding additional parameters with potentially different lengths means that the script will need to calculate the description length, add padding to get 4 byte alignment, etc.

Reply:
Agreed, its non-trivial to handle (I just barfed with exit 1 if it didnt fit the form 'x.y' exactly, with one digit for each